src/System.IO.Pipelines/tests/PipeReaderStreamTests.nonnetstandard.cs

```diff
@@ -0,0 +1,306 @@
+﻿using System;
+using System.Buffers;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace System.IO.Pipelines.Tests
+{
+    public class PipeReaderStreamTests
+    {
+        public delegate Task<int> ReadAsyncDelegate(Stream stream, byte[] data);
+
+        [Theory]
+        [MemberData(nameof(ReadCalls))]
+        public async Task ReadingFromPipeReaderStreamReadsFromUnderlyingPipeReader(ReadAsyncDelegate readAsync)
```

Comment:
You don't have tests that go async. Do we want any additional coverage in that area?

Reply:
Added